src/components/Quiz.js

```diff
@@ -18,19 +18,48 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    let checkedAnswer;
+    const answerId = Number(e.target.value);
+    const previousSelectedAnswers = answers[e.target.name] || [];
+    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
+      answerId,
+    );
+
+    if (isAnswerPreviouslySelected) {
+      const removePreviousAnswer = previousSelectedAnswers.filter(
+        answer => answer !== answerId,
+      );
+      checkedAnswer = {
+        [e.target.name]: removePreviousAnswer,
+      };
+    } else {
+      checkedAnswer = {
+        [e.target.name]: [...previousSelectedAnswers, answerId],
+      };
+    }
+    return checkedAnswer;
+  };
+
+  const handleAnswerSelect = (e, type) => {
+    let selectedAnswer;
+    const answerId = Number(e.target.value);
+    if (type === 'checkbox') {
+      selectedAnswer = handleCheckboxAnswers(e, userAnswers);
+    } else {
+      selectedAnswer = { [e.target.name]: [answerId] };
+    }
```

Comment:
```suggestion
    const answerId = Number(e.target.value);
    const selectedAnswer = type === 'checkbox'
      ? handleCheckboxAnswers(e, userAnswers)
      : { [e.target.name]: [answerId] };
```
Here we can convert the `if` statement to a ternary